### PR TITLE
Homework 11 JSON

### PR DESCRIPTION
**Pull request by nkstlrv**
There are many inputs in the program because I wanted to do the Phonebook more interactive.  
Moreover, this app was almost finished when you said during our class that we should avoid inputs.
Further will not use them, if not specified in the assignment.

### CODE REVIEW

File: HW_11/PhoneBook App/pb_functions/create.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        "Country": country
+    }
+
+    while True:
+        val_lst = []
+        key_lst = []
+        for key, val in new_cont.items():
+            if len(val_lst) == len(new_cont):
+                break
+            val_lst.append(val)
```
Comment (SergeyTovstonog) [inline HW_11/PhoneBook App/pb_functions/create.py:102]: not sure why you need that val_lst and key_lst.
you can print like
for key, value in new_cont.items():
    print(f"{key} -- {value}")

File: HW_11/PhoneBook App/pb_functions/create.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                  "\n Use only given numbers to choose")
+            continue
+        except Exception as error:
+            print(f"\n Unexpected error --> {error}"
+                  "\n Try again")
+            continue
+
+        if next_step == 1:
+
+            with open("HW_11\PhoneBook App\phonebook.json",
```
Comment (SergeyTovstonog) [inline HW_11/PhoneBook App/pb_functions/create.py:126]: you can have special function like save_json, as you repeat the same code in all functions

Comment (SergeyTovstonog) [inline HW_11/PhoneBook App/pb_functions/create.py:126]: I'd suggest to set constant PHONEBOOK_FILE = "HW_11\PhoneBook App\phonebook.json" and use it in all places.

File: HW_11/PhoneBook App/pb_functions/edit.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+                    continue
+
+                if next_step == 1:
+                    break
+                elif next_step == 2:
+                    quit()
+
+            contact_dict["Name"] = name
+            new_dict = contact_dict
+            phone_book1.remove(contact)
```
Comment (SergeyTovstonog) [inline HW_11/PhoneBook App/pb_functions/edit.py:83]: you may note the index of the item that needs to be edited and re-set the corresponding fields as well.

File: HW_11/PhoneBook App/pb_functions/search.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+with open(r"HW_11\PhoneBook App\phonebook.json", "r") as pb:
+    phone_book = json.load(pb)
+
+
+# Search by Name
+def search_name() -> str:
+    n_lst = []
+    while True:
+
+        name = input("\n Please enter contact's NAME: ").title()
```
Comment (SergeyTovstonog) [inline HW_11/PhoneBook App/pb_functions/search.py:12]: a lot of repeated code in the search function. 
check the example I've posted with `_search` function

File: HW_11/PhoneBook App/phonebook.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+# Program mode choice------------------------------------------------------------------------------------------
+while True:
+    prog_mode = int(input("\n No you need to choose the Program mode."
+                          "\n Here are the options:"
+                          "\n\n - Search --> [1]"
+                          "\n - Create new contact --> [2]"
+                          "\n - Edit contact's info --> [3]"
+                          "\n - Delete contact --> [4]"
+                          "\n - Quit the program --> [1101]"
```
Comment (SergeyTovstonog) [inline HW_11/PhoneBook App/phonebook.py:45]: quit could be something small, like 0 :)

**Review by SergeyTovstonog — APPROVED** [review]
Good implementation. A lot of repeated code, could be added to functions and reused

**Comment by nkstlrv** [comment]
Will work more on making the program more efficient and having less repeated code (Search mode especially)

**Pull request merged**